backend/pkg/api/data_access/user.go

```diff
@@ -6,19 +6,33 @@ import (
 	"math"
 	"time"
 
+	"github.com/gobitfly/beaconchain/pkg/api/services"
 	t "github.com/gobitfly/beaconchain/pkg/api/types"
 	"github.com/gobitfly/beaconchain/pkg/commons/utils"
 	"github.com/pkg/errors"
 )
 
 type UserRepository interface {
+	GetUserExists(email string) (bool, error)
+	AddUser(email, password string) error
 	GetUserCredentialInfo(email string) (*t.UserCredentialInfo, error)
 	GetUserIdByApiKey(apiKey string) (uint64, error)
 	GetUserInfo(id uint64) (*t.UserInfo, error)
 	GetUserDashboards(userId uint64) (*t.UserDashboardsData, error)
 	GetUserValidatorDashboardCount(userId uint64) (uint64, error)
 }
 
+func (d *DataAccessService) GetUserExists(email string) (bool, error) {
+	// TODO @patrick
+	return d.dummy.GetUserExists(email)
+}
+
+func (d *DataAccessService) AddUser(email, password string) error {
+	// TODO @patrick
+	d.services.QueueEmail(services.EMail{Recipient: email, Message: "Click to confirm email"}, 60*5)
```

Comment:
Sending the e-mail is not part of the data access and should be done inside the handler, since you're not fetching or manipulating data in our databases. This also means that the handler service needs direct access to the email service and this func is going to have to return some confirmation hash for the message though.